### PR TITLE
Added Modular Exponentiation Algorithm

### LINKED ISSUE #1006: [Feature]: Adding Modular Exponentiation Algorithm
### Feature Name

Modular Exponentiation Algorithm

### Feature Description

Adding Modular Exponentiation Algorithm

### Motivation

Adding Modular Exponentiation Algorithm

### Implementation Suggestions (Optional)

_No response_

### Feature Type

New Algorithm

### Does this feature require additional resources?

- [ ] Additional Documentation
- [ ] Third-Party Libraries
- [ ] UI/UX Design
- [X] Other

### References (Optional)

_No response_

### PR DESCRIPTION
**Pull request by shravya312**
### Description
added Added Modular Exponentiation Algorithm Introduction, Characteristics, working, Complexity analysis, example.

Fixes #1006 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by shravya312** [comment]
Hi @ajay-dhangar can u review the pr

File: docs/algorithms/modular-exponentiation-algorithm.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+- **Output**: The result of `(base^exp) % mod`, calculated efficiently.
+- **Constraints**: Large values for `base` and `exp` make a direct approach infeasible, so modular exponentiation is necessary for efficiency.
+
+## Solution Approach
+The **modular exponentiation algorithm** employs **exponentiation by squaring**, which reduces the problem size at each step by halving the exponent. This approach can be implemented recursively or iteratively and uses the following properties:
+
+- If `exp` is even:  
+  \[
+  \text{base}^{\text{exp}} = (\text{base}^{\text{exp}/2})^2
+  \]
```
Comment (ajay-dhangar) [inline docs/algorithms/modular-exponentiation-algorithm.md]: replace

```md
  \[
  \text{base}^{\text{exp}} = (\text{base}^{\text{exp}/2})^2
  \]
```

with

```md
$$
  \text{base}^{\text{exp}} = (\text{base}^{\text{exp}/2})^2
$$
```

Comment (shravya312) [reply docs/algorithms/modular-exponentiation-algorithm.md]: Sure I will make the changes @ajay-dhangar

**Comment by shravya312** [comment]
I have made the changes suggested can u review the changes @ajay-dhangar

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**